Modules/Scripted/DICOMPlugins/DICOMScalarVolumePlugin.py

```diff
@@ -224,10 +237,39 @@ def examineFiles(self, files):
             for tag in subseriesTags:
                 value = slicer.dicomDatabase.fileValue(file, self.tags[tag])
                 value = value.replace(",", "_")  # remove commas so it can be used as an index
+
                 if tag not in subseriesValues:
                     subseriesValues[tag] = []
-                if not subseriesValues[tag].__contains__(value):
+
+                if tag in vectorTags:
+                    if value != "":
+                        directionCosine = [float(i) for i in value.split("\\")]
+                        directionCosineA = numpy.array(directionCosine[:3])
+                        directionCosineB = numpy.array(directionCosine[-3:])
```

Comment:
The code is much more readable now!  In fact, now that I see what's going on I realize probably these called `rowDirectionCosines` and `columnDirectionCosines` for each better readability.

Reply:
But also I now realize that this probably won't work for `diffusionGradientOrientation` which has just 3 components. https://dicom.innolitics.com/ciods/enhanced-mr-color-image/enhanced-mr-color-image-multi-frame-functional-groups/52009229/00189117/00189076/00189089

So probably two branches are needed.

Reply:
I don't think we need to split to vectors at all, as we just compare each component anyway.

Reply:
right, the splitting was from the commit before (in which I was comparing the arrays by dot product). Not splitting and comparing all the components fix all. Thanks for catchign this. Fixed in https://github.com/Slicer/Slicer/pull/7362/commits/1ccb8ecb92cfb2c2a8db4d2cf5cdc46da5a2c04d